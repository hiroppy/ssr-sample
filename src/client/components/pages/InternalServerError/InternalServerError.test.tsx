import * as React from 'react';
import { shallow } from 'enzyme';
import { InternalServerError } from '.';

test('should call stopSaga', () => {
  const stop = jest.fn();

  shallow(<InternalServerError stopSaga={stop} />);

  expect(stop).toHaveBeenCalled();
});
