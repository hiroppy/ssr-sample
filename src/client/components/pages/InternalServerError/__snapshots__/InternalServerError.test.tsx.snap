// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render self and sub-components 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "testKey",
        "pathname": "/",
      },
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <InternalServerError>
      <Component
        title="500"
      >
        <HelmetWrapper
          defer={true}
          encodeSpecialCharacters={true}
        >
          <SideEffect(NullComponent)
            defer={true}
            encodeSpecialCharacters={true}
            link={
              Array [
                Object {
                  "href": "https://fonts.googleapis.com/css?family=Muli",
                  "rel": "stylesheet",
                },
              ]
            }
            meta={
              Array [
                Object {
                  "charset": "UTF-8",
                },
                Object {
                  "content": "width=device-width, initial-scale=1.0",
                  "name": "viewport",
                },
              ]
            }
            title="500"
            titleAttributes={Object {}}
          >
            <NullComponent
              defer={true}
              encodeSpecialCharacters={true}
              link={
                Array [
                  Object {
                    "href": "https://fonts.googleapis.com/css?family=Muli",
                    "rel": "stylesheet",
                  },
                ]
              }
              meta={
                Array [
                  Object {
                    "charset": "UTF-8",
                  },
                  Object {
                    "content": "width=device-width, initial-scale=1.0",
                    "name": "viewport",
                  },
                ]
              }
              title="500"
              titleAttributes={Object {}}
            />
          </SideEffect(NullComponent)>
        </HelmetWrapper>
      </Component>
      <Component>
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bdVaJa",
                  "isStatic": true,
                  "lastClassName": "cEKWjh",
                  "rules": Array [
                    "
  align-items: center;
  display: flex;
  flex-direction: column;
  margin: 15px 30px;

  & > *:not(:first-child) {
    margin-top: 24px;
  }
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bdVaJa",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sc-bdVaJa cEKWjh"
            >
              <p>
                Internal Server Error
              </p>
              <Link
                replace={false}
                to="/"
              >
                <a
                  href="/"
                  onClick={[Function]}
                >
                  Go Back to Top Page
                </a>
              </Link>
            </div>
          </StyledComponent>
        </styled.div>
      </Component>
    </InternalServerError>
  </Router>
</MemoryRouter>
`;
